src/Components/Cart.js

```diff
@@ -0,0 +1,182 @@
+import axios from "axios";
+import { useCart } from "../Contexts";
+import { wishlistHandler, checkIfAlreadyPresent,getFinalPrice } from "../utilities";
+
+export function Cart() {
+  const {
+    state: { cartItems, wishlistItems },
+    dispatch
+  } = useCart();
+
+  
+
+  async function removeFromCartHandler(item) {
+    try {
+      const response = await axios.post("./api/cartItems", {
+        cartItem: {
+          ...item,
+          existsInCart: false
+        }
+      });
+      if (response.status === 201) {
+        dispatch({ type: "REMOVE_FROM_CART", payload: item });
+      }
+    } catch (err) {
+      console.log(err);
+    }
+  }
+
+  function movetowishlist(item) {
+    removeFromCartHandler(item);
+    const response = checkIfAlreadyPresent(wishlistItems, item.id);
+    if (!response) {
+      wishlistHandler(wishlistItems, dispatch, item);
+    } else if (response && response.existsInWishlist === false) {
+      wishlistHandler(wishlistItems, dispatch, item);
+    }

```

Comment:
Just a good practice that you should use else at end of else if conditions. If wrong correct me,

Reply:
No Kunal . We should use an "else" only if we wish to do something if the " if " condn is not satisfied.
Here, we want to call wishlistHandler only if  any one of the two conditions is met. And if nothing is satisfied then we don't  want to do anything, hence I avoided writing "else" .

Reply:
No Kunal . We should use an "else" only if we wish to do something if the " if " condition is not satisfied.
Here, we want to call wishlistHandler only if any one of the two conditions is met. And if nothing is satisfied then we don't want to do anything, hence I avoided writing "else" .